### DISCUSSION TITLE
How to speed up `BAPI_USER_GET_DETAIL` calls.

### DISCUSSION THREAD

**Discussion by avhz** [SAP/PyRFC#373] [Q&A]
Hi :) 

I see that the project is possibly stagnating, however I would be grateful if someone can help me with speeding up `BAPI_USER_GET_DETAIL` calls.

I need to query many users (to retrieve email addresses), and it takes quite some time. 

My code is (roughly) the following. Is there a better way?

```python
def query_user(self, username: str) -> dict:
    data = self.connection.call(
        "BAPI_USER_GET_DETAIL",
        USERNAME=username,
    )
    return data
    
def query_user_email(self, username: str) -> str:
    data = self.query_user(username)["ADDRESS"]["E_MAIL"]
    return data
    
def query_user_emails(self, usernames: list) -> dict:
    emails = {usr: self.query_user_email(usr) for usr in usernames}
    return emails
```

**Comment by DevAfterSix** [comment]
I have an error when I call BAPI_USER_GET_DETAIL.
decimal.InvalidOperation: [<class 'decimal.ConversionSyntax'>]
Can you share the full code how to use dict or the document to learn more about dict.

**Reply by Rfolino501** [reply]
were you able to figure out the fix for this?
